### PR TITLE
Add support for delimiter value NIL in LIST responses

### LINKED ISSUE #1936: IMAP: Add support for delimiter value NIL in LIST response
### Expected behavior
K9 should list all IMAP folders. This worked OK with version 5.0, so I guess it is a regression.

### Actual behavior
K9 only lists the standard folders and then throws an exception.

### Steps to reproduce
1. IMAP server uw-imap
2. Refresh IMAP folders
3.

### Environment
K-9 Mail version: 5.201

Android version: 6.0.1

Account type (IMAP, POP3, WebDAV/Exchange): IMAP

IOException: Expected 0022 (") but got 004e (N)﻿

com.fsck.k9.mail.MessagingException: Unable to get folder list.
at com.fsck.k9.mail.store.imap.ImapStore.getPersonalNamespaces(ImapStore.java:156)
at com.fsck.k9.controller.MessagingController.refreshRemoteSynchronous(MessagingController.java:430)

From the log, it seems that it does get the folders before the error and I also do not see the actual exception in the log:
```
01-03 09:16:11.352 I/k9      (8350): Running command 'doRefreshRemote', seq = 7(foregroundpriority)
01-03 09:16:11.464 D/k9      (8350): Connecting to sidamos.de as sidamos.de/84.162.7.82
01-03 09:16:11.465 I/XiaomiFirewall(3882): firewall pkgName:com.fsck.k9, result:0x0
01-03 09:16:11.749 V/k9      (8350): conn265707776<<<#null# [OK, [CAPABILITY, IMAP4REV1, I18NLEVEL=1, LITERAL+, SASL-IR, LOGIN-REFERRALS, AUTH=PLAIN, AUTH=LOGIN], Backend.fritz.box IMAP4rev1 2007f.404 at Tue, 3 Jan 2017 09:16:08 +0100 (CET)]
01-03 09:16:11.749 D/k9      (8350): Saving [AUTH=LOGIN, LOGIN-REFERRALS, SASL-IR, IMAP4REV1, LITERAL+, I18NLEVEL=1, AUTH=PLAIN] capabilities for conn265707776
01-03 09:16:11.750 V/k9      (8350): conn265707776>>> 1 AUTHENTICATE PLAIN
01-03 09:16:11.807 V/k9      (8350): conn265707776<<<#+# []
01-03 09:16:11.875 V/k9      (8350): conn265707776<<<#1# [OK, [CAPABILITY, IMAP4REV1, I18NLEVEL=1, LITERAL+, IDLE, UIDPLUS, NAMESPACE, CHILDREN, MAILBOX-REFERRALS, BINARY, UNSELECT, ESEARCH, WITHIN, SCAN, SORT, THREAD=REFERENCES, THREAD=ORDEREDSUBJECT, MULTIAPPEND], User thomas authenticated]
01-03 09:16:11.875 D/k9      (8350): Saving [NAMESPACE, SCAN, WITHIN, BINARY, IMAP4REV1, ESEARCH, LITERAL+, MULTIAPPEND, MAILBOX-REFERRALS, IDLE, THREAD=REFERENCES, I18NLEVEL=1, SORT, UNSELECT, THREAD=ORDEREDSUBJECT, UIDPLUS, CHILDREN] capabilities for conn265707776
01-03 09:16:11.876 V/k9      (8350): conn265707776>>> 2 LIST "" "*"
01-03 09:16:11.945 V/k9      (8350): conn265707776<<<#null# [LIST, [\NoInferiors, \Marked], /, mythtv-dev]
01-03 09:16:11.945 V/k9      (8350): conn265707776<<<#null# [LIST, [\NoInferiors, \UnMarked], /, Entw&APw-rfe]
01-03 09:16:11.945 V/k9      (8350): conn265707776<<<#null# [LIST, [\NoInferiors, \UnMarked], /, eBay]

[...]

01-03 09:16:11.977 V/k9      (8350): LockableDatabase: Transaction ended, took 11ms / com.fsck.k9.mailstore.LocalFolder.appendMessages(LocalFolder.java:1218)
01-03 09:16:11.980 I/k9      (8350):  Command 'doRefreshRemote' completed
```

### PR DESCRIPTION
**Pull request by cketti**
Fixes #1936

### CODE REVIEW

File: k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapResponseParser.java [hunk 1]
```diff
@@ -412,6 +411,22 @@ private String parseQuoted() throws IOException {
         throw new IOException("parseQuoted(): end of stream reached");
     }
 
+    private String parseQuotedOrNil() throws IOException {
```
Comment (Valodim) [inline k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapResponseParser.java]: any reason why this one didn't get a `Nullable` annotation but the other method did?

Comment (cketti) [inline k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapResponseParser.java]: I didn't give it much thought. The other one is potentially public API, i.e. if we ever decide to release an IMAP library. It's also an exceptional case for these response data holder classes.

**Comment by cketti** [comment]
This is a regression in 5.2xx. So I want to include the fix in the next bugfix release.

**Comment by philipwhiuk** [comment]
Oh I thought next stable would mean 5.2x oops.

**Pull request merged**